### PR TITLE
Incomplete Implementation of Notification Channel

### LINKED ISSUE #312: [Assined] Incomplete Implementation of Notification Channel
#### Deprecated Constructors.
NotificationCompat.Builder(Context context) is deprecated. use NotificationCompat.Builder(Context, String) instead. 

####  When there is no notification, the setting of the notification channel does not appear.
I recommend that you execute createNotificationChannel() at the following timing.
* Application > onCreate()
* when received ACTION_LOCALE_CHANGED.

####  The name of NotificationChannel should not use appName.
Users can visit Settings to change the NotificationChannel behaviors.
In order to allow user judgment, functional names should be given.

## Links

https://developer.android.com/guide/topics/ui/notifiers/notifications.html#CreateChannel
https://developer.android.com/reference/android/app/NotificationChannel.html#NotificationChannel(java.lang.String, java.lang.CharSequence, int)
https://qiita.com/Shiozawa/items/095e77d38fc00681e898

### PR DESCRIPTION
**Pull request by nacatl**
## Issue
- close #312

## Overview (Required)
- (required) 
  - I replaced the deprecated constructor of NotificationBuilder with the recommended one.
  - The notification channel will be created when `Application.onCreated()` and received `Intent.ACTION_LOCALE_CHANGED`.
  - I renamed the channel name to `start favorite session(en)`/ `マイセッションの開始(ja)`.
- (optional)
  - For the good of type-safety, I created enum named `NotificationUtil.ChannelType`. 
  In addition, by this change, if a new channel type will be needed, just only creating an enum type will be required.

## Links
- https://developer.android.com/guide/topics/ui/notifiers/notifications.html#CreateChannel
- https://developer.android.com/reference/android/app/NotificationChannel.html#NotificationChannel(java.lang.String,%20java.lang.CharSequence,%20int)
- https://qiita.com/Shiozawa/items/095e77d38fc00681e898
- https://www.androidauthority.com/android-8-0-oreo-app-implementing-notification-channels-801097/

## Screenshot
- Notification setting screen(in the condition that the notification has been never shown)

Before | After(ja) | After(en)
:--: | :--: | :--:
312_before | 312_after_ja | 312_after_en

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -145,6 +145,11 @@
         </activity>
 
         <receiver android:name=".presentation.common.broadcastreceiver.NotificationBroadcastReceiver" />
+        <receiver android:name=".presentation.common.broadcastreceiver.LocaleChangedBroadcastReceiver">
```
Comment (takahirom) [inline app/src/main/AndroidManifest.xml]: 👍

File: app/src/main/java/io/github/droidkaigi/confsched2018/presentation/common/broadcastreceiver/LocaleChangedBroadcastReceiver.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import io.github.droidkaigi.confsched2018.util.initNotificationChannel
+import timber.log.Timber
+
+class LocaleChangedBroadcastReceiver : BroadcastReceiver() {
+    override fun onReceive(context: Context?, intent: Intent?) {
+        Timber.d("LocaleChangedBroadcastReceiver.onReceive")
+        if (Intent.ACTION_LOCALE_CHANGED == intent!!.action) {
```
Comment (takahirom) [inline app/src/main/java/io/github/droidkaigi/confsched2018/presentation/common/broadcastreceiver/LocaleChangedBroadcastReceiver.kt]: Probably we can use `intent?.action`. What do you think?

File: app/src/main/java/io/github/droidkaigi/confsched2018/util/NotificationUtil.kt [hunk 3]
```diff
@@ -5,40 +5,40 @@ import android.app.NotificationManager
...
 import android.support.annotation.RequiresApi
+import android.support.annotation.StringRes
 import android.support.v4.app.NotificationCompat
 import io.github.droidkaigi.confsched2018.R
+import io.github.droidkaigi.confsched2018.util.NotificationUtil.ChannelType
 
 class NotificationUtil {
-    companion object {
-        const val FAVORITE_SESSION_START_CHANNEL_ID = "favorite_session_start_channel"
+    enum class ChannelType(
```
Comment (takahirom) [inline app/src/main/java/io/github/droidkaigi/confsched2018/util/NotificationUtil.kt]: 👍

**Comment by takahirom** [comment]
Almost LGTM! It's 🆒

File: app/src/main/java/io/github/droidkaigi/confsched2018/presentation/common/broadcastreceiver/LocaleChangedBroadcastReceiver.kt (hunk 2 above)
Comment (nacatl) [reply app/src/main/java/io/github/droidkaigi/confsched2018/presentation/common/broadcastreceiver/LocaleChangedBroadcastReceiver.kt]: Oh, it is my shallowness...
I fix soon, so wait please a little.

Comment (nacatl) [reply app/src/main/java/io/github/droidkaigi/confsched2018/presentation/common/broadcastreceiver/LocaleChangedBroadcastReceiver.kt]: Fixed!

**Comment by takahirom** [comment]
LGTM :+1:
I want to pass you oreo :smile:

**Pull request merged**